tox.ini

```diff
@@ -1,22 +1,42 @@
 [tox]
-envlist = py27,py33,py34,docs,flake8
+envlist = py{34,35,36,37,38}, docs, flake8
+# On developers local machines usually not all of
+# python3.4...python3.8 interpreters are available, so
```

Comment:
Do you know that [pyenv](https://github.com/pyenv/pyenv) makes it really easy to fix that issue?

Reply:
Ah, no I hadn't heard of pyenv. From my brief investigation of pyenv I think it seems a bit more heavy-handed than I'd like, I don't think it would be worth it to add another dependency, and it also looks non-trivial to install, because I don't think it's available on PyPi. I think that local testing on a single version of python would catch the vast majority of bugs, and then the CI testing on all the versions is sufficient.

Reply:
I guess my comment was too laconic. This wouldn't be a dependency. This is only a convenient setup that can be recommended (not imposed) to contributors (not mere users). The workflow is: 
* install pyenv (this is a [one-liner](https://github.com/pyenv/pyenv-installer#installation--update--uninstallation), at least if you use a user-friendly OS)
* run `pyenv install 3.8.2` and the same with 3.7.2, 3.6.8 and 3.5.6 say.
* Put next to `tox.ini` a file `.python-version` containing:
   ```
    3.8.2
    3.7.2
    3.6.8
    3.5.6
    ```

That's all, you can now run `tox` in that folder and it will find all the required python versions. Also when you call `python` or `pip` inside that folder or a subfolder, it will use python 3.8.2 which is the first version mentioned above.